python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -584,6 +584,20 @@ def _prepare_for_get(self, ref: DatasetRef,
 
         if len(fileLocations) > 1:
             disassembled = True
+
+            # If trust is involved it is possible that there will be too
+            # many components listed here. We need to explicitly check
```

Comment:
I don't understand this first sentence.  Is it because some components appear in the execution butler because it has everything for the run but some components will not be written until a later Quantum execution?  The rest of the comment and the code makes sense.

Reply:
Registry has no concept of components so nothing in registry can tell me whether all components were written or only some of them. On read if you ask for a `calexp.wcs` it will know to look for exactly the `wcs` component so will fail or not depending on the presence of the file.